CIP-0149? | Optional DRep Compensation (#991)
cardano-foundation/CIPs: CIP-optional-drep-compensation/README.md

---
CIP: CIP-DRep-Comp
Title: Optional DRep Compensation
Category: Metadata
Status: Proposed
Authors:
    - Philip DiSarro <info@anastasialabs.com>
Implementors: []
Discussions:

>     - https://github.com/cardano-foundation/CIPs/pull/?

Reviewer: ```suggestion
    - https://github.com/cardano-foundation/CIPs/pull/991
```

## Motivation: why is this CIP necessary?
Introducing DRep compensation into the ledger will be a an absolutely massive undertaking, that will involve a large amount of engineering hours and a non-trivial price-tag. Also, the community has not yet reached consensus on whether DRep compensation is even desirable, or should be mandatory, or what the compensation scheme would even look like.

The best way to approach feature development is to release an MVP, evaluate its reception in the market, and use that data to determine the best way to proceed. This also allows us to introduce the feature without imposing it on anyone, it is opt-in, if you do not wish to compensate your DRep you can simply not opt-in. If, at any time, you wish to adjust the percentage of rewards that you are donating to your DRep you can just issue a new delegation certificate and adjust the amount or opt-out entirely.

## Specification
This feature is extremely simple to implement. When a user delegates to a DRep, the user should receive a prompt asking if they want to donate a percentage of their staking rewards to that DRep (ie. if they want to opt-in to compensate their DRep).

> If they choose to opt-in, they enter the percentage of their staking rewards that they would like their DRep to receive. The DRep delegation transaction will include that percentage in the transaction metadata in the following format:

Reviewer: ```suggestion
If they choose to opt-in, they enter the percentage of their staking rewards that they would like their DRep to receive. The DRep delegation transaction will include that percentage in the transaction metadata, with metadatum label `3692`, in   the following format:
```
add specific detail on `metadatum_label`, this PR should also this label to the CIP-10 registry to reserve it

> ```

Reviewer: ```suggestion
```json
```

{
    "3692": {

>         "int": PERCENTAGE_INTEGER_HERE 

Reviewer: ```suggestion
        "donationPercentage": PERCENTAGE_INTEGER_HERE 
```
might be nice to be more specific here

    }
}
```
Where the integer stored in the metadata is the thousandths place in the decimal so 1 would convey that the user is electing to donate 0.01% of their staking rewards to their DRep. This value should be obtained directly from the percentage the user described in the prompt (ie. the user put 0.5% in the prompt then the integer in the metadata would be 5).

> When a user claims their staking rewards, their wallet will look up their delegation transaction and determine if they have opted-in to compensate their DRep (by checking the tx metadata). If metadata in the format described above is found, then the wallet will add an additional output to the reward withdrawal transaction that sends the intended percentage of their rewards to their DRep. If there is not enough ada in their reward balance to pay the minAda for that output, the wallet will convey this to the user and let them decide how to proceed (or the user can establish how to handle this via wallet settings). 

Reviewer: ```suggestion
When a user claims their staking rewards, their wallet will look up their delegation transaction and determine if they have opted-in to compensate their DRep (by checking the tx metadata). If metadata in the format described above is found, then the wallet will add an additional output to the reward withdrawal transaction that sends the intended percentage of their rewards to their DRep. If there is not enough ada in their reward balance to produce a output which satisfies minimum `lovelacePerUTxOWord`, the wallet will convey this to the user and let them decide how to proceed (or the user can establish how to handle this via wallet settings). 
```

Reviewer: someone can correct me, but I believe it is `lovelacePerUTxOWord` that must be satisfied now

---

> CIP: CIP-DRep-Comp

Reviewer 2: ```suggestion
CIP: 149
```

Title: Optional DRep Compensation
Category: Metadata
Status: Proposed
Authors:
    - Philip DiSarro <info@anastasialabs.com>
Implementors: []
Discussions:
    - https://github.com/cardano-foundation/CIPs/pull/991
Created: 2025-02-19
License: CC-BY-4.0
---

## Abstract

> <!-- A short (\~200 word) description of the proposed solution and the technical issue being addressed. -->

Reviewer 2: ```suggestion
<!-- A short (\~200 word) description of the proposed solution and the technical issue being addressed. -->
```
(remove comment inherited from template scaffold)
